// jssip/lib/test/include/LoopSocket.js
"use strict";
// LoopSocket send message itself.
// Used P2P logic: message call-id is modified in each leg.
Object.defineProperty(exports, "__esModule", { value: true });
class LoopSocket {
    url = 'ws://localhost:12345';
    via_transport = 'WS';
    sip_uri = 'sip:localhost:12345;transport=ws';
    connect() {
        setTimeout(() => {
            this.onconnect();
        }, 0);
    }
    disconnect() { }
    send(message) {
        const new_message = this.modifyCallId(message);
        setTimeout(() => {
            this.ondata(new_message);
        }, 0);
        return true;
    }
    isConnected() {
        return true;
    }
    isConnecting() {
        return false;
    }
    onconnect() { }
    ondisconnect() { }
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    ondata(_event) { }
    // Call-ID: add or drop word '_second'.
    modifyCallId(message) {
        const begin = message.indexOf('Call-ID');
        const end = message.indexOf('\r', begin);
        let callId = message.substring(begin + 9, end);
        if (callId.endsWith('_second')) {
            callId = callId.substring(0, callId.length - 7);
        }
        else {
            callId += '_second';
        }
        return `${message.substring(0, begin)}Call-ID: ${callId}${message.substring(end)}`;
    }
}
exports.default = LoopSocket;
